upgrade/scripts/upgrade/util/upgrade-cray-nls.sh

```diff
@@ -50,38 +50,57 @@ function patchKeycloak() {
     --data-urlencode username="$(kubectl get secrets keycloak-master-admin-auth -o jsonpath='{.data.user}' -n services | base64 -d)" \
     --data-urlencode password="$(kubectl get secrets keycloak-master-admin-auth -o jsonpath='{.data.password}' -n services | base64 -d)" \
     https://api-gw-service-nmn.local/keycloak/realms/master/protocol/openid-connect/token | jq -r '.access_token')
-    #shellcheck disable=SC2126
-    foundArgoInRedirectUris=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .redirectUris' | grep "https://argo.cmn" | wc -l)
-    redirectUris=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .redirectUris')
-    #shellcheck disable=SC2126
-    foundArgoInWebOrigins=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins' | grep "https://argo.cmn" | wc -l)
-    webOrigins=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins')
 
-    id=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .id')
+    res_file=$(mktemp)
+    http_code=$(curl -s -o "${res_file}" -w "%{http_code}" -k -XGET -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients")
+    if [[ ${http_code} -ne 200 ]]; then
+        echo "Request Failed, Response code: ${http_code}"
+        cat "${res_file}"
+        exit 1
+    fi
+    
+    id=$(jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .id'  < "${res_file}" )
+    echo "Get client id: ${id}"
 
-    curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients/${id}" > client.json
-    if [[ ${foundArgoInRedirectUris} -ge 1 ]];then
-        echo "Argo URL is set in RedirectUris"
-        echo "${redirectUris}"
+    http_code=$(curl -s -o "client.json" -w "%{http_code}" -k -XGET -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients/${id}")
+    if [[ ${http_code} -ne 200 ]]; then
+        echo "Request Failed, Response code: ${http_code}"
+        cat "${res_file}"
```

Comment:
Is it res_file or client.json that you want to cat in this case?

Reply:
throw out bash, problem solved